edb/schema/inheriting.py

```diff
@@ -156,10 +156,24 @@ def inherit_fields(
             else:
                 ours = None
 
-            inherited = result is not None and ours is None
+            inherited = False
+            if field_name == 'cardinality':
```

Comment:
I'm not sure I like this approach of hardcoding logic for 'cardinality' specifically.  Perhaps there should be a `Field` attribute instead: `strict_inheritance` or something like that.

Reply:
Well, `cardinality` is already special. We have a cardinality-specific merge function (separate from, say, other special merges for  `readonly` and `required`). I'm not sure we gain anything here by adding a flag, but a generic flag will obscure what is actually being handled, I think.

Reply:
The reason why I bring up the cardinality-specific merge function is that it exists because cardinality has slightly different rules when it comes to inheritance, where you basically can't omit it like other things.

Reply:
`cardinality` is a field present only on `Pointer` objects.  It is weird to open-code for it in a generic piece of code in `inheriting.py`.  So, please either overload this in `pointers.py` or make a field attribute.

Reply:
Ah, OK, point taken about `inheriting.py` vs `pointers.py`. I'll overload `pointers.py` then.

Reply:
Inheritance has been refactored and overloaded in `pointers.py` to accommodate cardinality.